### DISCUSSION TITLE
Qiita CLI でタグに 'test' をセットすると ID 名の md が作成され二重投稿になる

### DISCUSSION THREAD

**Discussion by tettekete** [increments/qiita-discussions#945] [要望・アイデア・不具合]
以下は実際に発生した markdown の中身です。

`public/test-2024-12-23.md`:

```
---
title: test-2024-12-23
tags:
  - 'test'
private: false
updated_at: ''
id: null
organization_url_name: null
slide: false
ignorePublish: false
---
# new article body
```
 
この `public/test-2024-12-23.md` ファイルを `push` したところ以下の `public/05ada1cb70cfb52364b2.md` が作成され、両方が Qiita に公開されました。つまり二重投稿となりました。

```
---
title: test-2024-12-23
tags:
  - テスト
private: false
updated_at: '2024-12-23T15:55:46+09:00'
id: 05ada1cb70cfb52364b2
organization_url_name: null
slide: false
ignorePublish: false
---
# new article body
```

テスト投稿は迷惑になるため確認回数は 1 回のみですが、恐らく何らかのバグと思われます。

ユーザーとしても記事の削除時に身に覚えの無い `05ada1cb70cfb52364b2.md` もリポジトリから削除しなければならず、気付きにくく不便だと思います。

---
追記: その後他の記事でも発生しました。Front Matter の更新処理時に作成されたテンポラリファイルがコミットに含められてしまい publish 対象になっている感じがします。

**Comment by acronhub** [comment]
@tettekete 
ご報告ありがとうございます。

弊社で確認しましたが、同様の現象を確かめることができませんでした。
原因を特定するために、お手数ですが、以下の情報を教えていただけませんでしょうか🙏

- ローカル環境のQiita CLI のバージョン
  以下のコマンドでバージョンを確認できます。
  ```
  % npx qiita version
  ```

- Github Actionsのステップ「Run increments/qiita-cli/actions/publish@v1」の結果ログ
  以下の手順でログを確認できます。
  1. Github Actions の publish_articles ジョブを開く
  2. 「Run increments/qiita-cli/actions/publish@v1」のログを開く

**Reply by tettekete** [reply]
返信ありがとうございます。

> * ローカル環境のQiita CLI のバージョン
>   以下のコマンドでバージョンを確認できます。

```sh
$ npx qiita version
1.6.1
$ node --version
v20.15.1
```


> Github Actionsのステップ「Run increments/qiita-cli/actions/publish@v1」の結果ログ 以下の手順でログを確認できます。

二重投稿が発生したと思われる `push` 時の同ログは成功を示しているようです。

<img width="805" alt="スクリーンショット 2024-12-24 17 45 36" src="https://github.com/user-attachments/assets/ecb2256e-ef34-40d2-80ac-0bd098d3a415" />

このアクションにより GitHub Action 側で作成されたコミットは以下の様になっております。

```sh
$ git log -1 --name-status 2304345babf65d418de01441f8893a71bf89df9f
commit 2304345babf65d418de01441f8893a71bf89df9f
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Mon Dec 23 06:59:38 2024 +0000

    Updated by qiita-cli

A       public/05ada1cb70cfb52364b2.md
M       public/test-2024-12-23.md
```

関係あるかどうか分かりませんが、一つ前のコミットでタグに `''` と空のタグを設定して `push` し GitHub Actions が失敗しています。このミスを修正してコミット・プッシュしたときに、重複投稿が発生したようです。

コミットとプッシュの流れは以下の様になります。（ブランチ  `main` は「ローカルの `main`」と読み替えてください）

```mermaid
gitGraph TB:
 
    commit id: "タグが ''(空) のままコミット"
    branch "リモート(GitHub)"
    commit id: "Actions失敗" type: HIGHLIGHT
    checkout main
    commit id: "タグに 'test' をセットしてコミット"
    commit id: "他の記事のタグも修正"
    checkout "リモート(GitHub)"
    merge main
    commit id: "Actions成功" type: HIGHLIGHT
    commit id: "GitHub Acitons が成功し二重投稿コミットが作成される"
    checkout main
    merge "リモート(GitHub)"
```

**Reply by acronhub** [reply]
@tettekete
返信ありがとうございます。
いただいた情報から、調査をしておりますが、原因を特定するのに時間がかかりそうです🙇

原因が分かり次第、改めてご連絡させていただきます。

**Reply by acronhub** [reply]
@tettekete 
いただいた情報をもとに調査を進めたところ、下記が原因かもしれないと考えています。

- GitHub Actions の「Publish articles」ステップで記事が投稿されたが、その後の「Commit and push diff」ステップが実行されなかった
- そのため、投稿された記事が Qiita CLI で管理されていない状態になった

お手数ですが、失敗したジョブのログについて、もう少し詳細な情報を教えていただけますでしょうか。
具体的には、以下の内容について教えていただけると助かります。

- Run increments/qiita-cli/actions/publish@v1 の実行状況
  - 出力されているログに  (Posted: 記事ファイル名 -> 記事ID)の記載があるか知りたいです
- 失敗したステップの詳細なログ
  - どのようなエラーが発生して、失敗したのか、失敗した箇所を知りたいです

例えば、以下のようなキャプチャでいただけると助かります。

![github_actions_error](https://github.com/user-attachments/assets/4ebdf114-dc0f-4080-aa1b-bf8f7b13e0ff)

お手数をおかけいたしますが、どうぞよろしくお願いいたします🙏

**Reply by tettekete** [reply]
> Run increments/qiita-cli/actions/publish@v1 の実行状況
> * 出力されているログに  (Posted: 記事ファイル名 -> 記事ID)の記載があるか知りたいです

以下当該アクションのログスクリーンショットです。 「Posted: 記事ファイル名 -> 記事ID」の出力は出ているようですね。

<img width="1169" alt="image" src="https://github.com/user-attachments/assets/9261d749-804d-4b57-b3f2-6da7a6696fd0" />

**Reply by acronhub** [reply]
@tettekete 
返信ありがとうございます。
原因かもしれないと考えていたのが、二重投稿の原因であっていそうです。

以下について、教えてください :pray: 

1. ジョブが失敗した原因を調査するため、失敗したジョブに関連するコミットのmdファイルについて、詳細を教えていただけますでしょうか。
具体的には、test-2024-12-23.md というファイルと、他のmdファイルが存在しているかと思われます。
他のmdファイルになんらか問題があり失敗した可能性がありますので、 [サポートフォーム](https://support.qiita.com/hc/ja/requests/new) からファイルの内容について教えていただけますでしょうか。

2. 現在も二重投稿が発生していますでしょうか。

お手数をおかけしますが、よろしくお願いいたします :pray:

**Comment by acronhub** [comment]
@tettekete 
サポートフォームへ詳細のご連絡をいただき、ありがとうございました。

頂いた情報を元に調査させていただきましたが、再現できず原因を特定することができませんでした。
現在はtetteketeさんの方でも再発していないとのことですので、もしまた同じ現象が発生するなど困ったことがあればご連絡ください :pray:

ご不便をおかけしてしまい申し訳ありませんが、どうぞよろしくお願いいたします。
